game-core/src/main/java/games/strategy/triplea/delegate/BattleCalculator.java

```diff
@@ -717,16 +717,14 @@ public static CasualtyDetails selectCasualties(
       }
     }
     // check right number
-    if (!isEditMode
-        && !(numhits + damaged.size()
-            == (hitsRemaining > totalHitpoints ? totalHitpoints : hitsRemaining))) {
+    if (!isEditMode && !(numhits + damaged.size() == (Math.min(hitsRemaining, totalHitpoints)))) {
```

Comment:
This is really quite the conditional expression. The `Math.min` helps. Further refactoring would be good here, but I can certainly appreciate the improvement for now.